attribute.go

```diff
@@ -185,7 +190,8 @@ func (ad *AttributeDecoder) Next() bool {
 // Type returns the Attribute.Type field of the current netlink attribute
 // pointed to by the decoder.
 func (ad *AttributeDecoder) Type() uint16 {
-	return ad.attr().Type
+	// Mask off any flags stored in the high bits.
+	return ad.attr().Type & attrTypeMask
```

Comment:
Hiding a part of the original Type might be dangerous. I think, user should decide by themselves, if they want to see only parts of the Type.

Reply:
I don't think this is dangerous because these bits are used by the kernel to handle the Nested and NetByteOrder flags. If you want direct access to the full bits, you can use UnmarshalAttributes as normal.

Reply:
what about making this masking optional? Or providing second function without masking?

So far, it was handy to use `ad.Type()` in a switch statement and then cover the resulting types.

Reply:
I think my concern with exposing the flag bits at all is that the caller will have to remember to account for them, or make a duplicate case such as https://github.com/WireGuard/wgctrl-go/commit/dc235175870abf30dabaac70a81a99f479e99178#diff-2bfaf714bbb2315a74c7760755088e00L78.

What's the advantage to leaving them exposed? If you are implementing a genetlink interface, you probably have docs in front of you that indicate that a certain attribute contains nested attributes. We don't rely on the presence of the flag to switch to a different nested mode; it's all up to the caller to invoke ad.Nested and such as appropriate.

I just want to understand your use case. I would rather not add another function or an argument to mask or not.

Reply:
It would appear that libnl also masks off these bits: https://www.infradead.org/~tgr/libnl/doc/api/attr_8c_source.html#l00109.